src/ripple/app/misc/impl/AMMHelpers.cpp

```diff
@@ -153,6 +153,11 @@ adjustAmountsByLPTokens(
     std::uint16_t tfee,
     bool isDeposit)
 {
+    // If lpTokens contains a fractional part and at least 16 significant
+    // digits, it is possible that lpTokensActual is rounded down. This is to
+    // ensure that we don't create/burn more LPTokens (depending on
+    // whether it's AMMDeposit or AMMWithdraw transaction) than what the trader
+    // asked for.
```

Comment:
@seelabs  This is one of the points in the codebase where LPTokens are rounded down. Check the function above `adjustLPTokens`.

Consequently, `lpTokensActual` < `lpTokens` by a very small margin.